src/server/models/CategoryModel.js

```diff
@@ -0,0 +1,41 @@
+import mongoose from "mongoose";
+
+/**
+ * @swagger
+ * components:
+ *  schemas:
+ *    Category:
+ *      type: object
+ *      required:
+ *        - name
+ *        - backgroundColor
+ *        - photo
+ *      properties:
+ *        id:
+ *          type: string
+ *        name:
+ *          type: string
+ *        backgroundColor:
+ *          type: string
+ *        photo:
+ *          type: string
+ */
+
+const categorySchema = new mongoose.Schema({
+  name: { type: String, required: true },
+  backgroundColor: { type: String, required: true },
+  photo: { type: String, required: true },
+});
+
+categorySchema.set("toJSON", {
+  transform: (doc, ret) => {
+    ret.id = ret._id;
+    delete ret._id;
+    delete ret.__v;
+    return ret;
```

Comment:
interesting... kodėl trinamas _id? tas `__v` yra mongoDb naudojamas versijai, jį galima ir kitais būdais nuimti, per schemą `UserSchema = new mongoose.Schema({
    nickname: String,
    reg_time: {type: Date, default: Date.now}
}, {
    versionKey: false // You should be aware of the outcome after set to false
});`